File: src/test/java/nox/core/VersionTest.java
/**
 * Created by skol on 02.03.17.
 */
package nox.core;

import org.junit.Test;

import static org.junit.Assert.assertEquals;


public class VersionTest {

	@Test
	public void version_maxGtMin_success() {
		assertEquals(Version.MIN.compareTo(Version.MAX), -1);
	}

	@Test
	public void version_min_correct() {
		assertEquals(Version.min(Version.MAX, Version.DEFAULT), Version.DEFAULT);
		assertEquals(Version.min(Version.DEFAULT), Version.DEFAULT);
	}

	@Test
	public void version_max_correct() {
		assertEquals(Version.max(Version.DEFAULT, Version.MIN), Version.DEFAULT);
		assertEquals(Version.max(Version.DEFAULT, Version.MIN), Version.DEFAULT);
	}

	@Test
	public void version_parseFromString_withDashInBuild_correct() {
		assertEquals("5.4.2.SNAPSHOT", new Version("5.4.2-SNAPSHOT").toString());
		assertEquals("5.4.2.12356-5678", new Version("5.4.2_12356_5678").toString());
	}

}

File: src/test/kotlin/nox/core/VersionTest.kt
/**
 * Created by skol on 02.03.17.
 */
package nox.core

import org.junit.Assert.assertEquals
import org.junit.Test


class VersionTest {

	@Test
	fun version_maxGtMin_success() {
		assertEquals(Version.MIN.compareTo(Version.MAX).toLong(), -1)
	}

	@Test
	fun version_min_correct() {
		assertEquals(listOf(Version.MAX, Version.DEFAULT).min(), Version.DEFAULT)
		assertEquals(listOf(Version.DEFAULT).min(), Version.DEFAULT)
	}

	@Test
	fun version_max_correct() {
		assertEquals(listOf(Version.DEFAULT, Version.MIN).max(), Version.DEFAULT)
		assertEquals(listOf(Version.DEFAULT, Version.MIN).max(), Version.DEFAULT)
	}

	@Test
	fun version_parseFromString_withDashInBuild_correct() {
		assertEquals("5.4.2.SNAPSHOT", Version("5.4.2-SNAPSHOT").toString())
		assertEquals("5.4.2.12356-5678", Version("5.4.2_12356_5678").toString())
	}

}
